libraries/SITL/SIM_GPS.cpp

```diff
@@ -44,6 +44,36 @@ struct GPS_TOW {
     uint32_t ms;
 };
 
+// ensure the backend we have allocated matches the one that's configured:
+GPS_Backend::GPS_Backend(GPS &_front, uint8_t _instance)
+    : front{_front},
+      instance{_instance}
+{
+}
+
+ssize_t GPS_Backend::write_to_autopilot(const char *p, size_t size) const
+{
+    return front.write_to_autopilot(p, size);
+}
+
+ssize_t GPS_Backend::read_from_autopilot(char *buffer, size_t size) const
+{
+    return front.read_from_autopilot(buffer, size);
+}
+
+void GPS_Backend::update(const GPS_Data &d)
+{
+    if (_sitl == nullptr) {
+        _sitl = AP::sitl();
+        if (_sitl == nullptr) {
+            return;
+        }
+    }
+
+    update_read(&d);  // i.e. reading config from autopilot
```

Comment:
These comments were useful, could you instead put them in the header to document the method?

Reply:
Done